metabase/metabase
# [FE] Support multiple breakout positions

Epic https://github.com/metabase/metabase/issues/46536

Some internal refactoring to allow multiple breakouts for the same column:
- Changes `breakoutPosition` to `breakoutPositions`, like `filterPositions` in MBQL lib
- Omits `:unit` and `:binning` on `lib/breakoutableColumns` because there is no 1:1 mapping. Makes `breakoutColumn` pass these properties instead because for a given breakout there is always 1 column.
- Makes `lib/breakout` ignore duplicate breakouts, like in `lib/filter`
- Makes `lib/replace-clause` ignore attempts to create duplicate breakouts in the query
- Makes the FE use it and expect multiple breakouts for the same column. We still don't allow that in the UI.

How to verify if it's ready to be merged:
- CI should be green, no behavior changes

How to verify it implements the task:
- Remove `breakoutPositions.length === 0` check https://github.com/metabase/metabase/blob/f551e6f6019ca26dec4517e40bb2375ceaff75fb/frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.tsx#L114
- You would be able to create multiple breakouts for the same column; duplicate columns would be ignored, like we do for filters
- QP doesn't support this well in many cases, so we cannot really allow this now.
